tests/cli/test_utils.py

```
@@ -293,6 +294,71 @@ def fake_popen(*_args, **_kwargs) -> _FakeProcess:
+    _patch_run_services_environment(monkeypatch, tmp_path)
+    started: list[tuple[str, _FakeProcess]] = []
+
+    def fake_popen(command, **_kwargs) -> _FakeProcess:
+        proc = _FakeProcess(pid=1000 + len(started), polls_to_exit=1)
+        started.append((command[0], proc))
+        return proc
+
+    monkeypatch.setattr(utils.subprocess, "Popen", fake_popen)
+    monkeypatch.setattr(utils, "_wait_for_all_ready", lambda *_args: False)
+
+    with pytest.raises(RuntimeError, match="embedding.*ready"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)